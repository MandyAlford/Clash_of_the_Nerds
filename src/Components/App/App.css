.App {
  text-align: center;
  /* background-image: url('./assets/bg-greek.JPG');
  background-size: cover;
  background-position: center; */
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}
