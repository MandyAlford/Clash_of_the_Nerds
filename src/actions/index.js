export const addUser = (user) => ({
    type: 'ADD_USER',
    user
})

export const addTrivia = (trivia) => ({
    type: 'ADD_TRIVIA',
    trivia
})

export const addCorrectGuess = (guess) => ({
  type: 'ADD_CORRECT_GUESS',
  guess
})

export const addIncorrectGuess = (guess) => ({
  type: 'ADD_INCORRECT_GUESS',
  guess
})

export const resetAnswers = () => ({
  type: 'RESET_STORE_ANSWERS'
})

export const saveUsersTrivia = (trivia) => ({
  type: 'SAVE_TRIVIA',
  trivia
})
