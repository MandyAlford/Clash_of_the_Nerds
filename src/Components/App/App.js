import React, { Component } from 'react';
import './App.css';
import Login from '../Login/Login';
import Nerds from '../Nerds/Nerds';
import LoadingPage from '../LoadingPage/LoadingPage';
import TestContainer from '../TestContainer/TestContainer';
import { Route, Switch } from 'react-router-dom';
import { addTrivia } from '../../actions';
import { connect } from 'react-redux';

class App extends Component{
  constructor() {
    super()
    this.state = {
      user: {
        name: '',
        team: '20',
        level: 'easy'
      },
      chosen: '20',
      trivia: []
    }
  }
  fetchData = (team, level) => {
    console.log(team, level);
    fetch(`https://opentdb.com/api.php?amount=10&category=${team}&difficulty=${level}`)
      .then(response => response.json())
      .then(trivia => {
        this.setState({ trivia })
        this.props.loadTriviaToStore( trivia.results )
      })
  }
  addUser = (user) => {
    this.setState({ user })
    this.setState({ chosen: user.team })
  }
  render() {
    return (
      <div className='App'>
      <Switch>
        <Route exact path='/' render={() => {
          return (
            <section>
              <Login addUser={this.addUser} fetchData={this.fetchData}/>
              <Nerds />
            </section>
          )}}/>
        <Route exact path='/test' render={() => {
          return (
            <section>
              {!this.props.trivia.trivia.length && <LoadingPage />}
              {this.props.trivia.trivia.length &&<TestContainer trivia={this.props.trivia} />}
            </section>
          )}} />
      </Switch>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  loadTriviaToStore: (trivia) => { dispatch(addTrivia(trivia)) }
})

export const mapStateToProps = state => ({
  trivia: state
});

export default connect(mapStateToProps, mapDispatchToProps)(App)
