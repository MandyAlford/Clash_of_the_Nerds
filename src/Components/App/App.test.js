import React from 'react';
import { render } from '@testing-library/react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { addTrivia, addUser } from '../../actions';

describe('App', () => {

  describe('fetchData', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App />)
    })

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('mapStateToProps', () => {
      it('should be able to update state', () => {
        let mockState = [{
          category: "Mythology",
          type: "multiple",
          difficulty: "easy",
          question: "Which figure from Greek mythology traveled to the underworld to return his wife Eurydice to the land of the living?",
          correct_answer: "Orpheus",
          incorrect_answers: (3) ["Hercules", "Perseus", "Daedalus"],
          userGuess: "Perseus"
        },{
          category: "Mythology",
          type: "multiple",
          difficulty: "medium",
          question: "Hera is god of...",
          correct_answer: "Marriage",
          incorrect_answers: ["Agriculture", "Sea", "War"],
          userGuess: "Sea"
        }]
        const expected = {
          trivia: [{
            category: "Mythology",
            type: "multiple",
            difficulty: "easy",
            question: "Which figure from Greek mythology traveled to the underworld to return his wife Eurydice to the land of the living?",
            correct_answer: "Orpheus",
            incorrect_answers: (3) ["Hercules", "Perseus", "Daedalus"],
            userGuess: "Perseus"
          },{
            category: "Mythology",
            type: "multiple",
            difficulty: "medium",
            question: "Hera is god of...",
            correct_answer: "Marriage",
            incorrect_answers: ["Agriculture", "Sea", "War"],
            userGuess: "Sea"
          }],
        }
        const result = mapStateToProps(mockState)
        expect(result).toEqual(expected)
      })
    })

    describe('mapDispatchToProps', () => {
      it('should call dispatch with the trivia results action when loadTriviaToStore is called', () => {
        const mockDispatch = jest.fn();
        const trivia = [{
          category: "Mythology",
          type: "multiple",
          difficulty: "easy",
          question: "Which figure from Greek mythology traveled to the underworld to return his wife Eurydice to the land of the living?",
          correct_answer: "Orpheus",
          incorrect_answers: (3) ["Hercules", "Perseus", "Daedalus"],
          userGuess: "Perseus"
        },{
          category: "Mythology",
          type: "multiple",
          difficulty: "medium",
          question: "Hera is god of...",
          correct_answer: "Marriage",
          incorrect_answers: ["Agriculture", "Sea", "War"],
          userGuess: "Sea"
        }]
        const actionToDispatch = addTrivia(trivia)
        const mappedProps = mapDispatchToProps(mockDispatch)
        mappedProps.loadTriviaToStore(trivia)
        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
      })

      it('should call dispatch with the guest action when loadUserToStore is called', () => {
        const mockDispatch = jest.fn();
        const user = {
          name: 'Jessica Rabbit',
          team: '19',
          level: 'hard'
        }
        const actionToDispatch = addUser(user)
        const mappedProps = mapDispatchToProps(mockDispatch)
        mappedProps.loadUserToStore(user)
        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
      })

    })
  })
})
