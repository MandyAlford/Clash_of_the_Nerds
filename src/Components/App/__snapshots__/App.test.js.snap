// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App fetchData should match snapshot 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/test"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/account"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/results"
      render={[Function]}
    />
    <BadPathPage />
  </Switch>
</div>
`;
